tools/c7n_azure/c7n_azure/resources/key_vault_keys.py

```diff
@@ -0,0 +1,97 @@
+# Copyright 2018 Capital One Services, LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.from c7n_azure.provider import resources
+
+import logging
+
+from azure.keyvault.key_vault_id import KeyVaultId
+
+from c7n.filters import Filter
+from c7n.utils import type_schema
+
+from c7n_azure.provider import resources
+from c7n_azure.query import ChildResourceManager
+from c7n_azure.utils import ThreadHelper, ResourceIdParser
+
+
+log = logging.getLogger('custodian.azure.keyvault.keys')
+
+
+@resources.register('keyvault-keys')
+class KeyVaultKeys(ChildResourceManager):
+
+    class resource_type(ChildResourceManager.resource_type):
+        service = 'azure.keyvault'
+        client = 'KeyVaultClient'
+        enum_spec = (None, 'get_keys', {
+            'vault_base_url': lambda p: 'https://{0}.vault.azure.net'.format(p['name'])

```

Comment:
hmm.. its not a declarative spec that point.. its a method on the resource type. we do this as a method in a few places gcp providers (re static/class methods ala `get`), i'll likely add one to aws as well around arns.